modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-retrofit2/headerParams.mustache

```diff
@@ -1 +1 @@
-{{#isHeaderParam}}@Header("{{baseName}}") {{{paramName}}}: {{{dataType}}}{{/isHeaderParam}}
\ No newline at end of file
+{{#isHeaderParam}}@Header("{{baseName}}") {{{paramName}}}: {{{dataType}}}{{#required}}{{#defaultValue}} = {{{.}}}{{/defaultValue}}{{/required}}{{^required}}?{{#defaultValue}} = {{{.}}}{{/defaultValue}}{{^defaultValue}} = null{{/defaultValue}}{{/required}}{{/isHeaderParam}}
```

Comment:
@shanselm-ergon Hey, one question, this `{{#defaultValue}} = {{{.}}}{{/defaultValue}}` is the same as this `{{#defaultValue}} = {{{defaultValue}}}{{/defaultValue}}`?
Just to make sure because I didn't knew that sintaxe.

Reply:
@shanselm-ergon sorry if I didn't communicate clearly, I was not saying your code was incorrect, I was just asking to make sure I understood the code, because I didn't knew that sintaxe, but if it was correct, there was nothing wrong with it 🙂

Reply:
Yes, all good. I took the code from the issu #8809. Unfortunately I'm also not an expert in mustache. But I did some testing and it looks like both options work. In the mustache documentation I couldn't find a clear answer how the dot operator works. Therefore I switched back to the, at least for me, simpler syntax. 
I also executed the scripts again without any diff.
For me, this version is more readable, that was the reason I committed it.

Reply:
Thanks for taking the time to create this PR. Let's wait for CI to see if passes.